controller/state.go

```diff
@@ -325,7 +325,10 @@ func (m *appStateManager) CompareAppState(app *v1alpha1.Application, revision st
 			// * target resource not defined and live resource is extra
 			// * target resource present but live resource is missing
 			resState.Status = v1alpha1.SyncStatusCodeOutOfSync
-			syncCode = v1alpha1.SyncStatusCodeOutOfSync
+			// we don't apply to the application
+			if !resource.HasAnnotationOption(obj, common.AnnotationSyncStatusOptions, "Ignore") {
```

Comment:
Ignore isn't granular enough because it masks the situation where there are actual differences in the configmap or secret. This is why i proposed IgnoreExtra. Alternatively we could do:
`Ignore=extra`

Reply:
I'm missing something here. Is this better:

* If the resources is partially out of sync, then we don't want to ignore it. 
* If the resource is unexpected, i.e. Argo CD thinks we need to prune it, then we should ignore it completely.